File: src/main/java/com/example/Server_electronic_journale/model/GradeEntry.java
package com.example.Server_electronic_journale.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@ToString
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "grade_entries")
public class GradeEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "entry_id")
    private int entryId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "gradebook_id", nullable = false)
    private Gradebook gradebook;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subject_id", nullable = false)
    private Subject subject;

    @Column(nullable = false)
    private int grade;
}

File: app/src/main/java/com/example/electronic_journal/server/model/GradeEntry.kt
package com.example.electronic_journal.server.model

import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable

@Serializable
data class GradeEntry(
    @SerializedName("entry_id")
    val entryId: Int,

    @SerializedName("gradebook_id")
    val gradebookId: Int,

    @SerializedName("subject_id")
    val subjectId: Int,

    @SerializedName("grade")
    val grade: Int
)
